src/QAT/qat/purr/backends/verification.py

```diff
@@ -89,25 +101,77 @@ def get_verification_model(qpu_type: QPUVersion) -> Optional[VerificationModel]:
         )
 
     if qpu_type.make == Lucy.__name__:
-        # TODO: Should have an apply_setup_to_hardware in live which people can use to build our own architecture.
-        model = VerificationModel(qpu_type, LucyVerificationEngine)
-        raise NotImplementedError("No lucy-specific model created yet.")
+        return apply_setup_to_hardware(
+            hw= VerificationModel(qpu_type, LucyVerificationEngine),
+            qubit_count=8,
+            pulse_hw_x_pi_2_width=25e-9,
+            pulse_hw_zx_pi_4_width=3.18e-07,
+            pulse_measure_width=2.41e-06,
+        )
 
     return None
 
 
 class VerificationEngine(QuantumExecutionEngine, ABC):
-    def _execute_on_hardware(self, sweep_iterator, package: QatFile):
-        veri_list = list(package.meta_instructions)
-        veri_list.extend(package.instructions)
-        self.verify_instructions(veri_list)
+
+    def _execute_on_hardware(self, sweep_iterator, package: QatFile) -> bool:
+
+        #TODO: Need a discussion around this step. Where are instructions 'finalised'?
+        while not sweep_iterator.is_finished():
+            sweep_iterator.do_sweep(package.instructions)
+            position_map = self.create_duration_timeline(package)
```

Comment:
You're constantly overwriting the position map and will only pick up the final sweep, this bit needs looking at more. I'd like to see it not use a sweep at all, why do we need one here?